File: diziCafe-backend/src/main/java/com/diziCafe/service/UserService.java
package com.diziCafe.service;

import com.diziCafe.model.User;

import java.util.List;

public interface UserService {

    User getCurrentUser();

    List<User> getAllUsersIfAdmin();

    void deleteUserById(Long id);

    User updateUser(Long id, User updatedUser);

    void changePassword(Long userId, String oldPassword, String newPassword);

    User toggleBanUser(Long id);

}

File: dizi-cafe-frontend/src/services/userService.ts
import api from "./api"; // ✅ interceptor'lı axios instance

const BASE_URL = "http://localhost:8080/diziCafe/users";

// Oturum açmış kullanıcı bilgisi
export const getCurrentUser = async () =>
  api.get(`${BASE_URL}/me`);

// Kullanıcı güncelleme
export const updateUser = async (id: number, updatedUser: any) =>
  api.put(`${BASE_URL}/${id}`, updatedUser);

// Şifre değiştirme
export const changePassword = async (id: number, oldPassword: string, newPassword: string) =>
  api.put(`${BASE_URL}/${id}/change-password`, null, {
    params: { oldPassword, newPassword },
  });

// Admin: tüm kullanıcıları çekme (frontend’te şimdilik gerekmez ama ekledim)
export const getAllUsers = async () =>
  api.get(`${BASE_URL}`);

// Admin: kullanıcı silme
export const deleteUser = async (id: number) =>
  api.delete(`${BASE_URL}/${id}`);

export const toggleBanUser = (userId: number) =>
  api.put(`${BASE_URL}/${userId}/toggle-ban`);
